src/plugins/custom_import_map/public/components/vector_upload_options.tsx

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiFilePicker,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiText,
+  EuiSpacer,
+  EuiCard,
+  EuiFieldText,
+  EuiSelect,
+  EuiTextColor,
+  EuiFormRow,
+} from '@elastic/eui';
+import { getIndex, postGeojson } from '../services';
+import { FILE_PAYLOAD_SIZE, FILE_PAYLOAD_SIZE_IN_MB } from '../../common/constants/shared';
+import { toMountPoint } from '../../../../src/plugins/opensearch_dashboards_react/public';
+import { RegionMapOptionsProps } from '../../../../src/plugins/region_map/public';
+
+const VectorUploadOptions = (props: RegionMapOptionsProps) => {
+  const notifications = props.vis.notifications;
+
+  const INDEX_NAME_SUFFIX = '-map';
+  const INDEX_NAME_UPPERCASE_CHECK = /[A-Z]+/;
+  const INDEX_NAME_SPECIAL_CHARACTERS_CHECK = /[`!@#$%^&*()\=\[\]{};':"\\|,.<>\/?~]/;
+  const INDEX_NAME_NOT_BEGINS_WITH_CHECK = /^[-+_.]/;
+  const INDEX_NAME_BEGINS_WITH_CHECK = /^[a-z]/;
+  const INDEX_NAME_NOT_ENDS_WITH_CHECK = /.*-map$/;
+  const MAX_LENGTH_OF_INDEX_NAME = 250;
+  const GEO_SHAPE_TYPE = 'geo_shape';
+
+  const [value, setValue] = useState('');
+  const [isLoading, setLoading] = useState(false);
+  const [fileContent, setFileContent] = useState();
+
+  const onTextChange = (e) => {
+    setValue(e.target.value);
+  };
+
+  const onChange = (files) => {
+    if (files[0]) {
+      validateFileSize(files);
+      setFileContent(files);
+    }
+  };
+
+  const fetchElementByName = (elementName: string) => {
+    return document.getElementsByName(elementName)[0];
+  };
+
+  const validateIndexName = (typedIndexName: string, isIndexNameWithSuffix: boolean) => {
```

Comment:
Constructing a single error message as a concatenated string is definitely an antipattern. Take a look at EUI form validation an how to use the `error` and `isValid` props on forms and form controls. (Keeping an array of error messages instead of a single string would be preferable even if you were implementing validation from scratch).

Reply:
sounds good, updated!